pwndbg/dbg/lldb/__init__.py

```diff
@@ -191,6 +191,15 @@ def reg_write(self, name: str, val: int) -> bool:
                     # event.
                     self.proc.dbg._trigger_event(pwndbg.dbg_mod.EventType.REGISTER_CHANGED)
 
+                    # If we set the stack pointer, the inner object might have been invalidated, try
+                    # to restore it, as it should still be the selected frame.
+                    if (
+                        name in (reg_sets[pwndbg.aglib.arch.name].frame, "sp")
```

Comment:
Is the `"sp"` literal check needed here bcoz `frame` may be e.g. `"rsp"`? or maybe we have some alias which makes `regs.sp` always work even on different archs?

Reply:
It's the latter.